common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -181,6 +182,7 @@ def publish_event(event_name, **kwargs):
         if overlimit_block_keys:
             # Publish an event for analytics purposes:
             publish_event("removed", removed=format_block_keys(overlimit_block_keys), reason="overlimit")
+            previous_count = len(selected)
```

Comment:
Updating an in-scope variable like this (and then using that variable in the closure built by defining `publish_event` locally) is pretty confusing. It took me a moment to figure out what effect changing `previous_count` would have. It would be clearer if `publish_event` took that count as an argument, rather than pulling it from the parent scope.

Now that I look at it, same goes for `selected`.